homeassistant/components/sensor/sigfox.py

```diff
@@ -0,0 +1,140 @@
+"""
+Sensor for SigFox devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sigfox/
+"""
+import logging
+import datetime
+import json
+import voluptuous as vol
+import requests
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+API_URL = 'https://backend.sigfox.com/api/'
+CONF_API_LOGIN = 'api_login'
+CONF_API_PASSWORD = 'api_password'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_LOGIN): cv.string,
+    vol.Required(CONF_API_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the sigfox sensor."""
+    api_login = config[CONF_API_LOGIN]
+    api_password = config[CONF_API_PASSWORD]
+    sigfox = SigfoxAPI(api_login, api_password)
+    auth = sigfox.auth
+    devices = sigfox.devices
+
+    sensors = []
+    for device in devices:
+        sensors.append(SigfoxDevice(device, auth))
+    add_devices(sensors, True)
+
+
+def epoch_to_datetime(epoch_time):
+    """Take an ms since epoch and return datetime string."""
+    return datetime.datetime.fromtimestamp(epoch_time).isoformat()
+
+
+class SigfoxAPI(object):
+    """Class for interacting with the SigFox API."""
+
+    def __init__(self, api_login, api_password):
+        """Initialise the API object."""
+        self._auth = requests.auth.HTTPBasicAuth(api_login, api_password)
+        response = requests.get(API_URL + 'devicetypes', auth=self._auth)
```

Comment:
As an alternative could you use `helpers.aiohttp_client` to make the requests. E.g., 

```python
....
with async_timeout.timeout(10, loop=self.hass.loop):
    response = yield from self.websession.get(url)
    yield from response.text()
```

Add a timeout if you want to stick to `requests`.

Reply:
Added `timeout=10`